### PR TITLE
Date.toLocaleDateString() - add note that result depends on timezone

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes https://github.com/mdn/content/issues/10483

As requested in https://github.com/mdn/content/pull/10525

### CODE REVIEW

**Review by wbamberg — CHANGES_REQUESTED** [review]
Sorry to be a pain, but if we're going to add a generic comment here, it looks weird to have a specific comment at line 12 (now 13). So I suggest we should remove the " (varies according to default locale)" bit from line 12 (now 13)

File: live-examples/js-examples/date/date-tolocaledatestring.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 const event = new Date(Date.UTC(2012, 11, 20, 3, 0, 0));
-
 const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
 
+//NOTE: The "expected output" depends on timezone, and may not be correct on your browser
```
Comment (wbamberg) [inline live-examples/js-examples/date/date-tolocaledatestring.js:4]: "not be correct" is a bit strong.

```suggestion
//NOTE: Depending on your local timezone, your output may differ from "expected output"
```

**Comment by hamishwillee** [comment]
Thanks for review. I accepted your change.

> Sorry to be a pain, but if we're going to add a generic comment here, it looks weird to have a specific comment at line 12 (now 13). So I suggest we should remove the " (varies according to default locale)" bit from line 12 (now 13)

The specific comment is actually relevant and correct - ie the string format changes depending on locale settings, but only for the last case where you are using the default locale. We could do something like below, but I'm concerned people might not get the point that locale only affects the last case.

```
//NOTE: Depending on your local timezone and locale, your output may differ from "expected output"
```

And  now you see why I wanted to just update the doc content!

**Comment by wbamberg** [comment]
Then could we update all three comments:

```js
// expected output: Donnerstag, 20. Dezember 2012 (varies according to local timezone)

// expected output:  الخميس، ٢٠ ديسمبر، ٢٠١٢ (varies according to local timezone)

// expected output: Thursday, December 20, 2012 (varies according to local timezone and default locale)
```

**Comment by hamishwillee** [comment]
@wbamberg I did not think of that. I've done this, but with the comments like `// expected output (varies according to local timezone: Donnerstag, 20. Dezember 2012)`

This was mostly because in the middle case we switch from a left-to-right to a right-to-left language. Switching again seemed cruel. I also don't think it hurts to note the result might differ before showing it.

**Comment by hamishwillee** [comment]
This appears to be failing due to a change I have not made.

**Review by wbamberg — APPROVED** [review]
Thanks for making this change, looks great to me.

**Comment by wbamberg** [comment]
The warning is due to https://github.com/mdn/interactive-examples/pull/1782, because our version of ESLint doesn't support `hasIndices`. But it looks like the new version does! https://eslint.org/blog/2021/10/eslint-v8.0.0-released. so I guess we should upgrade.

**Comment by hamishwillee** [comment]
Thanks for the review and the help. Good to get this done.

**Comment by wbamberg** [comment]
Thank you for your patience with my nit-picking :)

**Comment by wbamberg** [comment]
Oh and I see we have https://github.com/mdn/interactive-examples/pull/1944 for the ESLint upgrade, it looks scary though!

**Pull request merged**